packages/tailwindcss-language-server/src/language/css-server.ts

```diff
@@ -0,0 +1,379 @@
+import {
+  getCSSLanguageService,
+  LanguageSettings,
+  DocumentContext,
+} from 'vscode-css-languageservice/lib/esm/cssLanguageService'
+import {
+  InitializeParams,
+  TextDocuments,
+  TextDocumentSyncKind,
+  WorkspaceFolder,
+  Disposable,
+  ConfigurationRequest,
+  CompletionItemKind,
+  Connection,
+} from 'vscode-languageserver/node'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { Utils, URI } from 'vscode-uri'
+import { getLanguageModelCache } from './languageModelCache'
+import { Stylesheet } from 'vscode-css-languageservice'
+import dlv from 'dlv'
+import { rewriteCss } from './rewriting'
+
+export class CssServer {
+  private documents: TextDocuments<TextDocument>
+  constructor(private connection: Connection) {
+    this.documents = new TextDocuments(TextDocument)
+  }
+
+  setup() {
+    let connection = this.connection
+    let documents = this.documents
+
+    let cssLanguageService = getCSSLanguageService()
+
+    let workspaceFolders: WorkspaceFolder[]
+
+    let foldingRangeLimit = Number.MAX_VALUE
+    const MEDIA_MARKER = '℘'
+
+    const stylesheets = getLanguageModelCache<Stylesheet>(10, 60, (document) =>
+      cssLanguageService.parseStylesheet(document),
+    )
+    documents.onDidOpen(({ document }) => {
+      connection.sendNotification('@/tailwindCSS/documentReady', {
+        uri: document.uri,
+      })
+    })
+    documents.onDidClose(({ document }) => {
+      stylesheets.onDocumentRemoved(document)
+    })
+    connection.onShutdown(() => {
+      stylesheets.dispose()
+    })
+
+    connection.onInitialize((params: InitializeParams) => {
+      workspaceFolders = (<any>params).workspaceFolders
```

Comment:
I don't think I've ever used a cast like that in TypeScript ever haha. Think I always do this:
```suggestion
      workspaceFolders = (params as any).workspaceFolders
```

Just a comment, no change required though.